modules/renter/download.go

```diff
@@ -700,7 +707,15 @@ func (dw *DownloadFileWriter) Destination() string {
 
 // WriteAt writes the passed bytes at the specified offset.
 func (dw *DownloadFileWriter) WriteAt(b []byte, off int64) (int, error) {
-	return dw.f.WriteAt(b, off-int64(dw.offset))
+	n, err := dw.f.WriteAt(b, off-int64(dw.offset))
+	if err != nil {
+		return n, err
+	}
+	dw.written += uint64(n)
+	if dw.written == dw.length {
+		return n, dw.f.Close()
+	}
```

Comment:
should probably be `dw.written >= dw.length`. Also, should we be truncated `b` so that we never write more than `dw.length` bytes total?

Reply:
Hmm, should we truncate, or return an error if writing `b` would exceed `length`?

Reply:
IIUC it's a developer error, so `build.Critical` would be appropriate.